### PR TITLE
[IGNORE] improve storybook decorator usage

### PR DESCRIPTION
**Pull request by juliepagano**
- Normalizes decorator parameter naming to be `lowerCamelCase`.
- Adds parameter typing decorators.
- Ensures decorators have unique names to reduce confusion and enable appropriate typing of parameters.
- Fixes some incorrect typings in stories with the help of the types

# Notes for reviewers

- This is a follow-up from https://github.com/perses/perses/pull/1159#discussion_r1191493889

# Checklist

- [x] Pull request has a descriptive title and context useful to a reviewer.
- [x] Pull request title follows the `[<catalog_entry>] <commit message>` naming convention using one of the following `catalog_entry` values: `FEATURE`, `ENHANCEMENT`, `BUGFIX`, `BREAKINGCHANGE`, `IGNORE`.
- [x] All commits have [DCO signoffs](https://github.com/probot/dco#how-it-works).

## UI Changes

- [x] Changes that impact the UI include screenshots and/or screencasts of the relevant changes.
- [x] Code follows the [UI guidelines](https://github.com/perses/perses/blob/main/ui/ui-guidelines.md).
- [x] Visual tests are stable and unlikely to be flaky. See [Storybook](https://github.com/perses/perses/tree/main/ui/storybook#visual-tests) and [e2e](https://github.com/perses/perses/tree/main/ui/e2e#visual-tests) docs for more details. Common issues include:
  - Is the data inconsistent? You need to mock API requests.
  - Does the time change? You need to use consistent time values or mock time utilities.
  - Does it have loading states? You need to wait for loading to complete.

### CODE REVIEW

File: ui/dashboards/src/components/Dashboard/Dashboard.stories.tsx [hunk 1]
```diff
@@ -223,7 +223,9 @@ function formatProviderParameters(dashboardState: DashboardResource) {
     },
     withTimeRange: {
       props: {
```
Comment (juliepagano) [inline ui/dashboards/src/components/Dashboard/Dashboard.stories.tsx:225]: Bad typing caught by the addition of the parameter typing. Really glad I decided to figure that out while I was in here.

File: ui/dashboards/src/components/Dashboard/Dashboard.stories.tsx [hunk 2]
```diff
@@ -532,16 +534,16 @@ const TIMESERIES_EXAMPLE_DASHBOARD_RESOURCE: DashboardResource = {
   },
 };
 
-const TIMESERIES_EXAMPLE_MOCK_NOW = 1673805600000;
-const TIMESERIES_EXAMPLE_MOCK_START = TIMESERIES_EXAMPLE_MOCK_NOW - 6 * 60 * 60 * 1000;
+const TIMESERIES_EXAMPLE_MOCK_END = 1673805600000;
```
Comment (juliepagano) [inline ui/dashboards/src/components/Dashboard/Dashboard.stories.tsx:537]: Decided to rename this while I was in here based on good feedback around similar mock times in the prior PR.

File: ui/dashboards/src/components/Dashboard/Dashboard.stories.tsx [hunk 3]
```diff
@@ -532,16 +534,16 @@ const TIMESERIES_EXAMPLE_DASHBOARD_RESOURCE: DashboardResource = {
...
+const TIMESERIES_EXAMPLE_MOCK_START = TIMESERIES_EXAMPLE_MOCK_END - 6 * 60 * 60 * 1000;
 export const ExampleWithTimeSeriesPanels: Story = {
   parameters: {
     ...formatProviderParameters(TIMESERIES_EXAMPLE_DASHBOARD_RESOURCE),
     withTimeRange: {
       props: {
         initialTimeRange: {
-          start: TIMESERIES_EXAMPLE_MOCK_START,
-          end: TIMESERIES_EXAMPLE_MOCK_NOW,
+          start: new Date(TIMESERIES_EXAMPLE_MOCK_START),
```
Comment (juliepagano) [inline ui/dashboards/src/components/Dashboard/Dashboard.stories.tsx:545]: These are supposed to be dates according to the types.

File: ui/dashboards/src/stories/decorators/WithDashboard.tsx [hunk 4]
```diff
@@ -15,6 +15,12 @@ import { StoryFn, StoryContext } from '@storybook/react';
 import { DashboardProvider, DashboardProviderProps } from '@perses-dev/dashboards';
 import { EMPTY_DASHBOARD_RESOURCE } from './constants';
 
+declare module '@storybook/react' {
```
Comment (juliepagano) [inline ui/dashboards/src/stories/decorators/WithDashboard.tsx:18]: This is the best way I could find to type the parameters for now based on looking at a few discussions/issues on the storybook github. It seems to work well as long as we don't have duplicates, which is why I renamed the small number of decorators with duplicate names.

**Review by sjcobb — APPROVED** [review]
🙌

**Pull request merged**